doc/manual/parallel-computing.rst

```diff
@@ -455,21 +455,131 @@ points: in this case, when :func:`remotecall_fetch` is called.
+- If a :class:`Channel` is empty, readers (on a :func:`take!` call) will block till data is available.
+- If a :class:`Channel` is full, writers (on a :func:`put!` call) will block till space becomes available.
+- :func:`isready` tests for the presence of any object in the channel, while :func:`wait`
+  waits for an object to become available.
+- A :class:`Channel` is in an open state initially. This means that it can be
+  read and written to freely via :func:`take!` and :func:`put!` calls. :func:`close` closes a :class:`Channel`.
+  On a closed :class:`Channel`\, :func:`put!` will fail. For example::
+
+        julia> c=Channel(2);
+
+        julia> put!(c,1);   # `put!` on an open channel succeeds
+
+        julia> close(c);
+
+        julia> put!(c,2);   # `put!` on a closed channel throws an exception.
+        ERROR: InvalidStateException("Channel is closed.",:closed)
+        in put!(::Channel{Any}, ::Int64) at ./channels.jl:77
+
+
+- :func:`take!` and :func:`fetch` on a closed channel successfully return any existing values till
+  it is emptied. Continuing the above example::
+
+        julia> take!(c)     # The first `take!` successfully returns available data.
+        1
+
+        julia> take!(c)
+        ERROR: InvalidStateException("Channel is closed.",:closed)
+        in wait(::Channel{Any}) at ./channels.jl:166
+
 
 A :class:`Channel` can be used as an iterable object in a ``for`` loop, in which
 case the loop runs as long as the :class:`Channel` has data or is open. The loop
 variable takes on all values added to the :class:`Channel`. An empty, closed :class:`Channel`
 causes the ``for`` loop to terminate.
 
+Consider a simple example using channels for inter-task communication.
+We start 4 tasks to process data from a single ``jobs`` channel.
+Jobs are identified by an id, a ``job_id``, which
+are written to the channel. Each task in this simulation reads a ``job_id``,
+waits for a random amout of time and writes back a tuple of ``job_id`` and the simulated time to
+the results channel. Finally all the ``results`` are printed out.
+
+.. _man-channels-example:
+
+.. doctest::
+    const jobs = Channel{Int}(32)
+    const results = Channel{Tuple}(32)
+
+    function do_work()
+        for job_id in jobs
+            exec_time = rand()
+            sleep(exec_time)                # simulates elapsed time doing actual work
+                                            # typically performed externally.
+            put!(results, (job_id, exec_time))
+        end
+    end
+
+    function make_jobs(n)
+        for i in 1:n
+            put!(jobs, i)
+        end
+    end
+
+    # feed the jobs channel with "n" jobs
+    n = 12
+    @schedule make_jobs(n)
+
+    # start 4 tasks to process requests in parallel
+    for i in 1:4
+        @schedule do_work()
+    end
+
+    # print out results
+    @elapsed while n > 0
+        job_id, exec_time = take!(results)
+        println("$job_id finished in $(round(exec_time,2)) seconds")
```

Comment:
Worth checking whether it actually runs if the prompts aren't there. I guess this might at least let us know if something broke in the API and the doc examples needed to be changed for it.